src/tir/analysis/calculate_workspace.cc

```diff
@@ -46,6 +48,10 @@ size_t WorkspaceCalculator::operator()(const PrimFunc& func) {
   return this->max_size;
 }
 
+size_t WorkspaceCalculator::GetByteAlignedSize(size_t non_aligned_size) {
+  return ((non_aligned_size + byte_alignment - 1) / byte_alignment) * byte_alignment;
```

Comment:
if we calculate alignment here, what happens if the workspace base is not itself aligned?

Reply:
That is expected to be aligned by user in the application.  E.g.,

`  __attribute__((section("SRAM1"), aligned(16))) static uint8_t workspace_buffer [WORKSPACE_SIZE];
   extern tvm_model_t network;
   tvm_workspace_t app_workspace;
       int main(...) {
           ...
           StackMemoryManager_Init(&app_workspace, workspace_buffer, WORKSPACE_SIZE);
           tvm_runtime_run(&network, inputs, outputs);
       }`

Reply:
okay sure, but also since this calculator only does Workspace memory, what's the plan for when we later fuse graph and workspace memory planning such that they could use the same memory pool? it seems like the workspace memory alignment should be a property of the function that allocates it, no?

Reply:
Yes, that is right -- in that work we'll just directly encode the offsets within the buffer that can respect the alignment. Until that we can have a global (sub-optimal) alignment accross all tensors accessed. Does that make sense?

Reply:
okay, i think that makes sense to me. i do wonder if it makes sense to upstream this given we're just going to change it later, but given the target parameter does not have a default, it's probably fine to do this as an interim solution until we address the broader memory planning challenges.

Reply:
Well; I've made sure it defaults to 1 wherever its accessed, so should not affect anything else if its not specified.
This enables us to see where we are in-terms of memory usage (by actually running models using the calculated value) -- that motivates the unified memory planning work even more.